thinc/backends/ops.py

```diff
@@ -680,6 +682,167 @@ def backprop_relu(
+        dX = slope * ((-2.5 < X) & (X < 2.5))
+        if inplace:
+            dY *= dX
+            return dY
+        return dY * dX
+
+    def swish(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        if inplace:
+            X *= self.sigmoid(X)  # type: ignore
+            return cast(FloatsType, X)
+        out = X * self.sigmoid(X)  # type: ignore
+        return cast(FloatsType, out)
+
+    def backprop_swish(
+        self, dY: FloatsType, X: FloatsType, Y: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        Y = Y + self.sigmoid(X) * (1 - Y)  # type: ignore
+        if inplace:
+            dY *= Y  # type: ignore
+            return cast(FloatsType, dY)
+        out = dY * Y  # type: ignore
+        return cast(FloatsType, out)
+
+    # Following https://www.scitepress.org/Papers/2019/74696/74696.pdf
+    def hard_swish(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        if inplace:
+            X *= self.hard_sigmoid(X)  # type: ignore
+            return cast(FloatsType, X)
+        out = X * self.hard_sigmoid(X)  # type: ignore
+        return cast(FloatsType, out)
+
+    def backprop_hard_swish(
+        self, dY: FloatsType, X: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        dX = X * 0.4 + 0.5
+        dX[X > 2.5] = 1.0
+        dX[X < -2.5] = 0
+        if inplace:
+            dY *= dX
+            return dY
+        return dY * dX
+
+    # From https://arxiv.org/pdf/1905.02244v5.pdf
+    def hard_swish_mobilenet(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        if inplace:
+            X *= self.relu_n(X + 3) / 6
+            return X
+        return X * (self.relu_n(X + 3) / 6)
+
+    def backprop_hard_swish_mobilenet(
+        self, dY: FloatsType, X: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        dX = (1 / 6) * (X * 2.0 + 3.0)
+        dX[X > 3.0] = 1.0
+        dX[X < -3.0] = 0
+        if inplace:
+            dY *= dX
+            return dY
+        return dX * dY
+
+    # https://newbedev.com/is-there-an-easily-available-implementation-of-erf-for-python
+    def erf(self, X: FloatsType) -> FloatsType:
+        # save the sign of x
+        sign = self.xp.sign(X)
+        X = self.xp.abs(X)
+
+        a1 = 0.254829592
+        a2 = -0.284496736
+        a3 = 1.421413741
+        a4 = -1.453152027
+        a5 = 1.061405429
+        p = 0.3275911
+
+        t = 1.0 / (1.0 + p * X)
+        y = 1.0 - (((((a5 * t + a4) * t) + a3) * t + a2) * t + a1) * t * self.xp.exp(
+            -X * X
+        )
+        return sign * y
+
+    def sechsq(self, X: FloatsType) -> FloatsType:
+        return (1 / self.xp.cosh(X)) ** 2
+
+    # https://github.com/huggingface/transformers/blob/master/src/transformers/activations.py#L37
+    def gelu_approx(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        tmp = 1.0 + self.xp.tanh(
+            self.xp.sqrt(2 / math.pi) * (X + 0.044715 * self.xp.power(X, 3))
```

Comment:
Better to make this a constant:

`self.xp.sqrt(2 / math.pi)`

To avoid recomputing it over and over again.

Reply:
I wasn't sure where is the best place to store this constant. Do you have a suggestion?

Reply:
I think it's good custom to place constants somewhere near the top of the file. (Probably between the typevars and the `Ops` class?)

Reply:
okok, I was debating whether i should make it an attribute of something or is it just better to put it as an actual constant. Thanks!